oviewer/convert_es.go

```diff
@@ -169,193 +181,292 @@ func (es *escapeSequence) convert(st *parseState) bool {
 	return false
 }
 
-// csToStyle returns tcell.Style from the control sequence.
-func csToStyle(style tcell.Style, params string) (tcell.Style, error) {
-	if params == "0" || params == "" || params == ";" {
-		return tcell.StyleDefault.Normal(), nil
+// sgrStyle returns tcell.Style from the SGR control sequence.
+func sgrStyle(style tcell.Style, paramStr string) tcell.Style {
+	if paramStr == "0" || paramStr == "" || paramStr == ";" {
+		return tcell.StyleDefault.Normal()
 	}
 
-	if s, ok := csiCache.Load(params); ok {
+	if s, ok := sgrCache.Load(paramStr); ok {
 		style = applyStyle(style, s.(OVStyle))
-		return style, nil
+		return style
 	}
 
-	s, err := parseCSI(params)
-	if err != nil {
-		return style, err
-	}
-	csiCache.Store(params, s)
-	return applyStyle(style, s), nil
+	s := parseSGR(paramStr)
+	sgrCache.Store(paramStr, s)
+	return applyStyle(style, s)
 }
 
-// parseCSI actually parses the style and returns ovStyle.
-func parseCSI(params string) (OVStyle, error) {
+// parseSGR actually parses the style and returns ovStyle.
+func parseSGR(paramStr string) OVStyle {
 	s := OVStyle{}
-	fields := strings.Split(params, ";")
-	for index := 0; index < len(fields); index++ {
-		field := fields[index]
-		num, err := toESCode(field)
+	paramList := strings.Split(paramStr, ";")
+	for index := 0; index < len(paramList); index++ {
+		sgr, err := toSGRCode(paramList, index)
 		if err != nil {
-			if errors.Is(err, ErrNotSuuport) {
-				return s, nil
-			}
-			return s, err
+			return OVStyle{}
 		}
-		switch num {
-		case 0:
+
+		switch sgr.code {
+		case 0: // Reset.
```

Comment:
Use constants for the numbers, this way code is clearer and you improve readability, because then you don't need comment


```suggestion
		case sgrCodeReset:
```

Reply:
I disagree with this.
Writing escape sequences with (uncommon) names makes them difficult to read.